### DISCUSSION TITLE
Line width (map)

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#351] [Q&A] [answered]
I've been looking at OpenAndroMaps Elevate/Elements themes for a few days now. I like them more and more, but they have the disadvantage (for me) that all the roads are drawn as very thin. Currently we can influence the map font size and icon size, but not everything else.

It's surely included somewhere in the theme files, but finding where exactly is probably impossible. Hence the question: is it possible to add something like a map size factor setting (as it is, for example, in Locus)?

**Comment by devemux86** [comment]
@Ciutnik thanks for the suggestion.

Mapsforge / VTM themes support the `base-stroke-width` tag to adjust the lines.

e.g. add `base-stroke-width="1.5"` at the top next to `map-background` tag.

---

Note: Elevate theme is created / tested on Mapsforge engine.
It is not a native VTM (OpenGL) theme, like the Cruiser themes.

It works in VTM (OpenGL) engine via translations (not always perfectly).

**Reply by Ciutnik** [reply]
> add base-stroke-width="1.5" at the top next to map-background tag.

Thank you very much! It works! 👍

However, I have to be careful because the effects can be... troublesome 😁

**Reply by Ciutnik** [reply]
Can a similar setting be applied to street names as well? By default they are tiny, much smaller than e.g. in Biker theme. Of course I can set `Font size (map)` but it also affects the font size of the labels above the points.

**Reply by devemux86** [reply]
> Can a similar setting be applied to street names as well?

The font size setting is global on the map.

Unless you change the theme and then all changes are possible. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
What I meant was to change the size of all texts in the Elevate theme, without a global change in Cruiser.

**Reply by devemux86** [reply]
Can use the `base-text-scale` or `base-text-size` in the theme similarly to the `base-stroke-width`.

**Reply by Ciutnik** [reply]
Thank you again! You are great! 👍

The solution turned out to be simple, but I certainly wouldn't have found it in the extensive documentation 🙂

_EDIT: I'm ashamed, it's clearly written at the __very beginning__ of the documentation! Why didn't I think of that? Because it's darkest under the lamp?_ 😉

**Comment by devemux86** [comment]
It should be possible to also offer an option for "Line width" in Cruiser settings.

As we already have settings for "Font size" and "Icon size",
it is more flexible to have them separate, instead of a single map size option.

**Comment by devemux86** [comment]
For reference:
- VTM [theme documentation](https://github.com/mapsforge/vtm/blob/master/docs/Rendertheme.md) and [xml schema](https://github.com/mapsforge/vtm/blob/master/resources/rendertheme.xsd)
- Mapsforge [theme documentation](https://github.com/mapsforge/mapsforge/blob/master/docs/Rendertheme.md) and [xml schema](https://github.com/mapsforge/mapsforge/blob/master/resources/renderTheme.xsd)

**Comment by devemux86** [comment]
"Line width (map)" setting will be available in the next Cruiser:

<img src="https://github.com/devemux86/cruiser/assets/3484020/3cec4421-df67-48dd-9393-07cca01814aa" width="50%"/>

**Reply by Ciutnik** [reply]
Great information! Thanks!

When experimenting with Elevate themes, your tips helped me achieve the perfect result, but it required excluding the thickening of sidewalks and pedestrian paths. I suspect the Cruiser's new line width setting will affect everything, including sidewalks and paths (as long as they are on the map)?

**Reply by devemux86** [reply]
It is a scaling of all the lines on the map.
As each theme has different rules, it is difficult to process only a subset of map elements.

**Reply by Ciutnik** [reply]
> It is a scaling of all the lines on the map.

Sure, it's obvious and expected 🙂

**Comment by Ciutnik** [comment]
Cool. Thank you for implementing 👍
However, I have a small request related to this. On my setup, the best setting is 200%. This is already the limit without any reserve. Maybe you could increase the limit to e.g. 300% just in case? 

By the way, the same applies to the font size on the map. To make it clearly visible, I had to set it to 200%, which is the maximum. You could probably increase all 200% limits to e.g. 300% 🙂

**Reply by devemux86** [reply]
@Ciutnik these are easy to change. :slightly_smiling_face: 

Do you also mean the font size (map) or also the icon size (map / favorites)?

**Reply by Ciutnik** [reply]
Using OpenAndroMaps and the Elevate theme, I had to set 200% for lines (map), icons (map POIs) and font (map), i.e. everything related to the map/theme. Perhaps the font size could be even slightly larger, hence my request to increase all 200% limits.

The size of my Favorites is currently 130% and that's enough for me, but it doesn't depend on the map or theme, so (for me) the 200% limit is enough here. However, maybe it would be consistent if it was the same as the other related limits?

**Reply by devemux86** [reply]
I will increase the (map) size limits.

Let's keep the favorites size until we need it, as they are auto-resized anyway from screen density.
(like all other overlays, e.g. waypoints icon size)

**Reply by Ciutnik** [reply]
I remembered why I didn't want to use font resizing on the map until now! Unfortunately, this affects the font size in the descriptions and labels of our points, tracks and routes. If you set it to 200% you can't see it...

Maybe there should be separate settings for elements included on the map (names of streets, cities, POI) and our map elements (Favorites, start/stop points, turning points, etc.).

Edit: I just noticed that in b15 the font sizes of "our" map elements are no longer related to the elements contained in the map 🙂

**Reply by devemux86** [reply]
> I just noticed that in b15 the font sizes of "our" map elements are no longer related to the elements contained in the map

Indeed, the settings marked with "(map)" should only affect the map.
The overlays like markers and lines are rendered separately, already are DPI optimized, so better have their own settings.

Currently all overlay lines have settings (routes, tracks). Favorites + waypoints also have their own settings.
We could think for size setting for the other markers, if we need it.

**Comment by devemux86** [comment, marked as answer]
> "Line width (map)" setting will be available in the next Cruiser

Available in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
